### PR TITLE
Fixed warnings that arise from stricter compiler flags

### PR DESCRIPTION
**Pull request by dellaert**
These are a number of small, mostly inconsequential edits that triggered warnings in a rather strict (but still not -Werror) environment.

### CODE REVIEW

File: cmake/GtsamBuildTypes.cmake [hunk 1]
```diff
@@ -130,6 +130,8 @@ else()
     $<$<CXX_COMPILER_ID:GNU>:-Wreturn-local-addr -Werror=return-local-addr>            # Error: return local address
     $<$<CXX_COMPILER_ID:Clang>:-Wreturn-stack-address   -Werror=return-stack-address>  # Error: return local address
     $<$<CXX_COMPILER_ID:Clang>:-Wno-misleading-indentation>  # Eigen triggers a ton!
+    $<$<CXX_COMPILER_ID:Clang>:-Wno-weak-template-vtables>  # TODO(dellaert): don't know how to resolve
+    $<$<CXX_COMPILER_ID:Clang>:-Wno-weak-vtables>  # TODO(dellaert): don't know how to resolve
```
Comment (gchenfc) [inline cmake/GtsamBuildTypes.cmake:134]: Can you mention which file these occurred in, just for reference?

**Review by gchenfc — APPROVED** [review]
Lgtm

File: cmake/GtsamBuildTypes.cmake (hunk 1 above)
Comment (dellaert) [reply cmake/GtsamBuildTypes.cmake:134]: Don’t remember all, but basically it occurred in several .cpp files where a template was explicitly instantiated.

**Comment by dellaert** [comment]
Thanks @gchenfc !!

**Pull request merged**